src/sentry/integrations/github/integration.py

```diff
@@ -458,67 +492,80 @@ def get_app_url(self) -> str:
-                status=ObjectStatus.PENDING_DELETION,
-            ).exists()
-
-        if integration_pending_deletion_exists:
-            return error(
-                request,
-                self.active_organization,
-                error_short="GitHub installation pending deletion.",
-                error_long=ERR_INTEGRATION_PENDING_DELETION,
+        with record_event(IntegrationPipelineViewType.GITHUB_INSTALLATION).capture() as lifecycle:
+            installation_id = request.GET.get(
+                "installation_id", pipeline.fetch_state("installation_id")
             )
+            if installation_id is None:
+                return self.redirect(self.get_app_url())
 
-        try:
-            # We want to limit GitHub integrations to 1 organization
-            installations_exist = OrganizationIntegration.objects.filter(
-                integration=Integration.objects.get(external_id=installation_id)
-            ).exists()
-
-        except Integration.DoesNotExist:
-            return pipeline.next_step()
-
-        if installations_exist:
-            return error(
-                request,
-                self.active_organization,
-                error_short="Github installed on another Sentry organization.",
-                error_long=ERR_INTEGRATION_EXISTS_ON_ANOTHER_ORG,
+            pipeline.bind_state("installation_id", installation_id)
+            self.determine_active_organization(request)
+            lifecycle.add_extra(
+                "organization_id",
+                self.active_organization.organization.id if self.active_organization else None,
             )
 
-        # OrganizationIntegration does not exist, but Integration does exist.
-        try:
-            integration = Integration.objects.get(
-                external_id=installation_id, status=ObjectStatus.ACTIVE
-            )
-        except Integration.DoesNotExist:
-            return error(request, self.active_organization)
-
-        # Check that the authenticated GitHub user is the same as who installed the app.
-        if (
-            pipeline.fetch_state("github_authenticated_user")
-            != integration.metadata["sender"]["login"]
-        ):
-            return error(
-                request,
-                self.active_organization,
-                error_short="Authenticated user is not the same as who installed the app",
-            )
+            integration_pending_deletion_exists = False
+            if self.active_organization:
+                # We want to wait until the scheduled deletions finish or else the
+                # post install to migrate repos do not work.
+                integration_pending_deletion_exists = OrganizationIntegration.objects.filter(
+                    integration__provider=GitHubIntegrationProvider.key,
+                    organization_id=self.active_organization.organization.id,
+                    status=ObjectStatus.PENDING_DELETION,
+                ).exists()
```

Comment:
The diffs are a bit weird for this change. Is the only difference the context manager and `record_failure` lifecycle calls?

Reply:
yes unfortunately the indentation makes it looks like all of this is new. only the context manager, `record_failure`, `add_extra` calls are new

Reply:
It's crazy 😭 I'm surprised github haven't figured out indent yet.